doc/boot_storage_needed_info.md

```diff
@@ -0,0 +1,38 @@
+List of Information Needed About Storage Configuration
+======================================================
+
+Mount Points
+---------------
+
+- what partition is mounted to "/" ( so it can be queried )
+- what partition is mounted to "/boot" and if it is same as "/" ( so it can be queried )
+- what partition is mounted to "/boot/zipl" ( only on s390 to query )
+- what partition is mounted to "/boot/efi" ( only on efi based archs )
+
+Partitions
+---------------
+
+- is partition logical ( so MBR have to be installed to extended one )
+- extended partition for given disk ( needed only if /boot is on logical partition )
+- file system of partition ( as some fs have problem embedding stage1 code, so do not propose to install it there )
+- fsid of partition ( e.g. to detect bios_grub partition and others )
+- if partition is part of MD, DM or Linux RAID ( target map now returns it, needed to decide if partition can be used for stage1 code )
+- raid level ( needed to check if it is 1, where we support redundancy boot from )
+- if partition is freshly created or deleted ( well, for deleted is to not count partitions that will gone in installation, for creation it is used for prep partition chooser, as newly created prep is prefered as it will not overwrite existing code )
+- if partition have boot flag ( bootloader ensure at least one partition have it and if requested, then also mark partition as one for boot )
+- marking partition with boot flag ( currently done directly via parted, but in future it would be nice to have it in one location, marking partition means also ensuring that no other partition have it )
+- where virtual partitions like EVMS, LVM, RAID ones, etc. layes on physical devices ( in some cases it have to use physical devices to write boot code )
+- convert udev device to kernel name and also kernel name to udev name according to settings in storage ( mountby option ) ( needed for persistant storage names feature )
```

Comment:
for udev, I usually get it from storage. Regarding alternative name, it was used by autoyast profile. That I will need check where it is actually translated. But I know that e.g. for uuid translation is done in bootloader_finish phase, when all mapping is available.